### PR TITLE
Create VS Code extension post

### PR DESCRIPTION
**Pull request by rebeccadee**
Asana task: https://app.asana.com/0/1199959077348208/1201686011425865/f

### CODE REVIEW

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+socialImage: https://sourcegraphstatic.com/blog/vs-code-extension/sourcegraph-vs-code-extension.png
+published: true
+---
+
+![Explore millions of open source repositories right from your IDE](https://sourcegraphstatic.com/blog/vs-code-extension/sourcegraph-vs-code-extension.png)
+
+My name is [Murat Sutunc](https://handbook.sourcegraph.com/team#murat-sutunc), and I’m the Engineering Manager for the Growth and Integrations team at Sourcegraph. Our team currently owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our IDE and browser extensions), and Sourcegraph extensions. As someone who does a lot of code reviews and occasionally jumps in to fix small issues, I wanted to share with everyone how I use my favorite IDE (VS Code!) and Sourcegraph together.
+
+TL;DR 👉 The Sourcegraph VS Code extension is a really powerful tool that lets you search your code and 2M+ open source repositories, all from your IDE! It’s available on [the marketplace](https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph)—no account needed, give it a try!
```
Comment (rebeccadee) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:17]: Wondering if we should give this some kind of different style treatment, e.g. like at the top of this post: https://about.sourcegraph.com/blog/log4j-log4shell-0-day/?

@muratsu @iskyOS

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+### Finding examples to fix issues
+
+Now and then, I jump in to help our team with some issues that are not on the critical path. For those unfamiliar, I think it's fair to say Sourcegraph's codebase is sizeable. It still fits on my laptop hard drive, but it’s large enough to tire IDEs when indexing. On top of our large main repository, we have several hundred additional repositories, and quickly finding what I'm looking for can be challenging.
```
Comment (rebeccadee) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:21]: @muratsu Could we link to our codebase to demonstrate what you mean by sizeable? I'm sure people will be curious as you say it!

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+<video loop autoplay muted playsinline>
+  <source src="https://sourcegraphstatic.com/vs-code-extension/vs-code-regex-usecase.webm" type="video/webm" />
+  <source src="https://sourcegraphstatic.com//vs-code-extension/vs-code-regex-usecase.mp4" type="video/mp4" />
+</video>
+
+Using this regex and the Sourcegraph Extension, I was able to find the code I’m looking for without leaving my IDE and get distracted by tons of other tabs that I have open in Chrome.
+
+### Researching an error
```
Comment (rebeccadee) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:38]: @iskyOS I wonder if we should retitle this section to align better to the use case? 
```suggestion
### Researching and fixing vulnerabilities
```

**Review by rebeccadee — COMMENTED** [review]
@iskyOS @muratsu Please take a last look to make sure everything looks as expected!

File: blogposts/2022/sourcegraph-extension-for-vs-code.md (hunk 1 above)
Comment (iskyOS) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:17]: @rebeccadee I like calling out that section! We could definitely use that alert box style.

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+I ran the following search query to find all Sourcegraph extensions we have (repository name pattern is sourcegraph-extension-name) which make use of the rxjs library:
+
+```
+repo:^github\.com/sourcegraph/sourcegraph-.+$  /import .+ from 'rxjs'/
+```
+
+<video loop autoplay muted playsinline>
+  <source src="https://sourcegraphstatic.com/vs-code-extension/vs-code-regex-usecase.webm" type="video/webm" />
+  <source src="https://sourcegraphstatic.com//vs-code-extension/vs-code-regex-usecase.mp4" type="video/mp4" />
+</video>
```
[the remark this answers is not in the document]
Comment (iskyOS) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:34]: @rebeccadee It looks like no files exist at those URLs — I can try moving them to GCP

[the remark this answers is not in the document]
Comment (rebeccadee) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:34]: @iskyOS I did upload them there but I may have used the wrong URL structure 🤦‍♀️ Let me see what I can do. Thanks!

[the remark this answers is not in the document]
Comment (iskyOS) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:34]: Oh sorry missed your later message! OK, sounds good.

**Comment by rebeccadee** [comment]
@jdorfman 👋 This is the blog post we're publishing tomorrow, FYI!

File: blogposts/2022/sourcegraph-extension-for-vs-code.md (hunk 2 above)
Comment (iskyOS) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:23]:
```suggestion
Now and then, I jump in to help our team with some issues that are not on the critical path. For those unfamiliar, I think it's fair to say [Sourcegraph's codebase is sizeable](https://github.com/sourcegraph/sourcegraph). It still fits on my laptop hard drive, but it’s large enough to tire IDEs when indexing. On top of our large main repository, we have several hundred additional repositories, and quickly finding what I'm looking for can be challenging.
```

Comment (muratsu) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:23]: Can we link to https://github.com/sourcegraph instead? I think the fact that we have 530 repositories is also important. Not only a med size primary codebase but also bits of code sits in multiple repos.

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+```
+repo:^github\.com/sourcegraph/sourcegraph-.+$  /import .+ from 'rxjs'/
+```
+
+<video loop autoplay muted playsinline>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-extension/vs-code-regex-usecase.webm" type="video/webm" />
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/vs-code-extension/vs-code-regex-usecase.mp4" type="video/mp4" />
+</video>
+
+Using this regex and the Sourcegraph Extension, I was able to find the code I’m looking for without leaving my IDE and get distracted by tons of other tabs that I have open in Chrome.
```
Comment (iskyOS) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:38]: This sentence reads a little funny to me with present/past tense — @rebeccadee is this grammatically correct?

**Comment by iskyOS** [comment]
@rebeccadee It feels like it ends a little bit abruptly — is it alright if I add a little bit of content to the closing paragraph? I can take an attempt at it, and if you don't think it fits / detracts from it, we can toss it.

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+As you can see in the video below, the global search returns several repositories that require further investigation. I finally changed the search context to Sourcegraph to narrow down my search scope to find all the repositories affecting our organization.
+
+<video loop autoplay muted playsinline>
+  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.webm" type="video/webm" />
+  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.mp4" type="video/mp4" />
+</video>
+
+I hope that you find these workflows useful. If you have questions or suggestions, you can reach out to me on Twitter ([@muratsutunc](http://twitter.com/muratsutunc)). You can [download the Sourcegraph VS Code extension from this link](https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph). Happy hacking!
```
Comment (iskyOS) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:55]:
```suggestion
I hope that you find these workflows useful. The extension has become a handy part of my day-to-day to stay in flow without leaving my IDE, especially for finding code to reuse or to quickly find and resolve vulnerabilities. Plus, the extension allows you to search Sourcegraph's index of millions of OSS repositories (without needing an account), so you can search for code examples, libraries, and best practices without ever leaving your IDE.


If you have questions or suggestions, you can reach out to me on Twitter ([@muratsutunc](http://twitter.com/muratsutunc)). You can [download the Sourcegraph VS Code extension from this link](https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph). Happy hacking!
```

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+My name is [Murat Sutunc](https://handbook.sourcegraph.com/team#murat-sutunc), and I’m the Engineering Manager for the Growth and Integrations team at Sourcegraph. Our team currently owns the initial user journey, growth-oriented projects, our code host and third-party integrations (including our IDE and browser extensions), and Sourcegraph extensions. As someone who does a lot of code reviews and occasionally jumps in to fix small issues, I wanted to share with everyone how I use my favorite IDE (VS Code!) and Sourcegraph together.
+
+<div class="alert alert-primary" role="alert" style="max-width:650px; align-items: center; margin: auto auto 20px auto">
+      <span>TL;DR 👉 The Sourcegraph VS Code extension is a really powerful tool that lets you search your code and 2M+ open source repositories, all from your IDE! It’s available on <a = href="https://marketplace.visualstudio.com/items?itemName=sourcegraph.sourcegraph">the marketplace</a>—no account needed, give it a try!</span>
+</div>
+
+### Finding examples to fix issues
```
Comment (iskyOS) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:21]:
```suggestion
### Finding reusable code to fix issues
```

File: blogposts/2022/sourcegraph-extension-for-vs-code.md (hunk 5 above)
Comment (iskyOS) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:38]:
```suggestion
Using this regex and the Sourcegraph Extension, I was able to find the code I’m looking for without leaving my IDE and get distracted by tons of other tabs that I have open in Chrome. I can quickly reuse that code to solve my issue without reinventing the wheel, too.
```

Comment (iskyOS) [reply blogposts/2022/sourcegraph-extension-for-vs-code.md:38]: (Adding some clarity that this is about code resuse)

**Review by iskyOS — APPROVED** [review]
Left a handful of comments / small additions to focus on use cases and tie back to open source search as well.

**Review by muratsu — APPROVED** [review]
Everything looks great!

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
+---
+title: 'How I use the Sourcegraph extension for VS Code'
+description: 'Sourcegraph’s extension for Visual Studio Code allows you to search millions of open source repositories without cloning them to your local machine or leaving your IDE. Here’s how one engineer integrates it into his workflow.'
```
Comment (rebeccadee) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:3]:
```suggestion
description: 'Sourcegraph’s extension for Visual Studio Code allows you to search millions of open source repositories without cloning them to your local machine or leaving your IDE. Here’s how one engineering manager integrates it into his workflow.'
```

File: blogposts/2022/sourcegraph-extension-for-vs-code.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+As you can see in the video below, the global search returns several repositories that require further investigation. I finally changed the search context to Sourcegraph to narrow down my search scope to find all the repositories affecting our organization.
+
+<video loop autoplay muted playsinline>
+  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.webm" type="video/webm" />
+  <source src="https://storage.cloud.google.com/sourcegraph-assets/blog/vs-code-extension/vscode-log4j.mp4" type="video/mp4" />
+</video>
+
+I hope that you find these workflows useful. The extension has become a handy part of my day-to-day to stay in flow without leaving my IDE, especially for finding code to reuse or to quickly find and resolve vulnerabilities. Plus, the extension allows you to search Sourcegraph's index of millions of OSS repositories (without needing an account), so you can search for code examples, libraries, and best practices without ever leaving your IDE.
```
Comment (rebeccadee) [inline blogposts/2022/sourcegraph-extension-for-vs-code.md:55]:
```suggestion
I hope you find these workflows useful. The extension has become a handy part of my day-to-day flow, especially for finding code to reuse or to quickly find and resolve vulnerabilities without leaving my IDE. Plus, the extension allows you to search Sourcegraph's index of millions of OSS repositories (without needing an account), so you can search for code examples, libraries, and best practices without ever leaving your IDE.
```

**Comment by rebeccadee** [comment]
Thanks @iskyOS and @muratsu for the feedback! We're gtg 🚢

**Pull request merged**